### PR TITLE
Bump runc to b2567b37d7b75eb4cf325b77297b140ea686ce8f

### PR DESCRIPTION
**Pull request by vdemeester**
This commit removes the cycle dependency between runc and docker

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

### CODE REVIEW

File: vendor/archive/tar/common.go [hunk 1]
```diff
@@ -1,286 +0,0 @@
-// Copyright 2009 The Go Authors. All rights reserved.
```
Comment (vdemeester) [inline vendor/archive/tar/common.go]: 🤔 @tiborvass I think I missed sthg here 😓

**Comment by cyphar** [comment]
It looks like a bunch of `_test` files were committed for some reason?

**Comment by vdemeester** [comment]
@cyphar yeah and that `vendor/archive/tar` thing is weird too 😓 
@tiborvass @tonistiigi help 😛

**Comment by thaJeztah** [comment]
> It looks like a bunch of _test files were committed for some reason?

Yes, I noticed that on another PR as well; those seem to have been added through https://github.com/moby/moby/pull/34895 (the `vendor` validation wasn't triggered on that one)

**Comment by cyphar** [comment]
@vdemeester I reckon just doing

```
rm -rf vendor/
vndr
```

Would fix the issue.

**Comment by tonistiigi** [comment]
@vdemeester `vndr` can't fetch the stdlib overrides but it was able to remove the unnecessary files so I don't know what was done in #34895 . If `archive/tar` needs to be updated there are instructions how it was created at the end of `vendor.conf` but there are no reasons for having any changes in there unless you are updating golang. If you don't remove that directory manually `vndr` will just ignore it. 

All this is based on `vndr` that was used when `archive/tar` was added. If there have been updates to it after that then things might have changed.

**Comment by tonistiigi** [comment]
Seems #35283 already cleaned up the extra files from #34895

**Comment by thaJeztah** [comment]
ah, yes, that was the one...guess I got confused by the archive files

**Comment by thaJeztah** [comment]
This bump was taken care of by https://github.com/moby/moby/pull/35465

**Pull request closed without merging**